locator/tablets.hh

```diff
@@ -216,6 +218,18 @@ public:
         return _tablets;
     }
 
+    /// Calls a given function for each tablet in the map in token ownership order.
+    template <typename Func>
+    requires std::invocable<Func, tablet_id, const tablet_info&>
+    future<> for_each_tablet(Func func) const {
```

Comment:
noncopyable_function? let's avoid bloat. Also lets us put this in .cc.

Reply:
Ok